File: OnlineGroceryOrderingSystemAuthenticationJava/src/main/java/com/alpha/ogo/service/impl/RoleServiceImpl.java
package com.alpha.ogo.service.impl;

import java.util.List;
import java.util.Optional;

import com.alpha.ogo.domain.Role;
import com.alpha.ogo.dto.CommonResponce;
import com.alpha.ogo.repo.RoleRepo;
import com.alpha.ogo.service.RoleService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;


/**
 * @author Tharindu
 *
 * Jan 4, 2021
 */

@Service
public class RoleServiceImpl implements RoleService {
	
	@Autowired
	private RoleRepo roleRepo;

	@Override
	public CommonResponce<Role> createRole(Role role) {
		CommonResponce<Role> resopnce = new CommonResponce<Role>();
		try {
			role.setRoleCode(role.getRoleCode().toUpperCase());
			Role roleObj=roleRepo.save(role);
			
			resopnce.setObject(roleObj);
			resopnce.setCode(CommonResponce.successCode);
			resopnce.setMessage(CommonResponce.successMessage);
			
		} catch (Exception e) {
			resopnce.setCode(CommonResponce.errorCode);
			resopnce.setMessage(CommonResponce.errorMessage);
		}
		
		return resopnce;
	}
	
	@Override
	public CommonResponce<Role> getAllRole() {
		CommonResponce<Role> resopnce = new CommonResponce<Role>();
		try {
			
			List<Role> roleObjList=roleRepo.findAll();
			
			resopnce.setList(roleObjList);
			resopnce.setCode(CommonResponce.successCode);
			resopnce.setMessage(CommonResponce.successMessage);
			
		} catch (Exception e) {
			resopnce.setCode(CommonResponce.errorCode);
			resopnce.setMessage(CommonResponce.errorMessage);
		}
		
		return resopnce;
	}

	@Override
	public CommonResponce<Role> getRoleById(Role role) {
		CommonResponce<Role> resopnce = new CommonResponce<Role>();
		try {
			
			Optional<Role> roleObj=roleRepo.findById(role.getRoleId());
			
			resopnce.setObject(roleObj.get());
			resopnce.setCode(CommonResponce.successCode);
			resopnce.setMessage(CommonResponce.successMessage);
			
		} catch (Exception e) {
			resopnce.setCode(CommonResponce.errorCode);
			resopnce.setMessage(CommonResponce.errorMessage);
		}
		
		return resopnce;
	}

	@Override
	public CommonResponce<Role> getrolebyRoleCode(String role) {
		CommonResponce<Role> resopnce = new CommonResponce<Role>();
		try {
			
			Role roleObj=roleRepo.getrolebyRoleCode(role);
			
			resopnce.setObject(roleObj);
			resopnce.setCode(CommonResponce.successCode);
			resopnce.setMessage(CommonResponce.successMessage);
			
		} catch (Exception e) {
			resopnce.setCode(CommonResponce.errorCode);
			resopnce.setMessage(CommonResponce.errorMessage);
		}
		
		return resopnce;
	}

	@Override
	public CommonResponce<Role> deleteRole(Role role) {
		CommonResponce<Role> resopnce = new CommonResponce<Role>();
		try {
			
			Optional<Role> roleObj=roleRepo.findById(role.getRoleId());
			roleRepo.delete(roleObj.get());
			
			resopnce.setCode(CommonResponce.successCode);
			resopnce.setMessage(CommonResponce.successMessage);
			
		} catch (Exception e) {
			resopnce.setCode(CommonResponce.errorCode);
			resopnce.setMessage(CommonResponce.errorMessage);
		}
		
		return resopnce;
	}

}

File: OnlineGroceryOrderingSystemAuthentication/src/main/kotlin/com/allion/ogoss/service/impl/RoleServiceImpl.kt
package com.allion.ogoss.service.impl

import com.allion.ogoss.domain.Role
import com.allion.ogoss.dto.CommonResponce
import com.allion.ogoss.mapper.RoleMapper
import com.allion.ogoss.service.RoleService
import com.allion.ogoss.util.ControllerHelper
import org.springframework.stereotype.Service


/**
 * @author  Tharindu
 * @date  6/2/2021 6:39 PM
 * @version 1.0
 */

@Service
class RoleServiceImpl (private var roleRepo : RoleMapper) : RoleService ,ControllerHelper()  {

    override fun createRole(role: Role?): CommonResponce<Role?>? {
        val response = CommonResponce<Role?>()
        try {
            if(role?.roleId==null){
                role?.roleCode=role?.roleCode?.toUpperCase()
                roleRepo.save(role as Role)
            }else{
                roleRepo.update(role)
            }

            response.newObject=role
            response.code = successCode
            response.message = successMessage
        } catch (e: Exception) {
            println(e)
            response.code = errorCode
            response.message = errorMessage
        }
        return response
    }

    override fun getAllRole(): CommonResponce<Role?>? {
        val response = CommonResponce<Role?>()
        try {
            val roleObjList: List<Role>? = roleRepo.findAll()
            response.list = roleObjList
            response.code = successCode
            response.message = successMessage
        } catch (e: java.lang.Exception) {
            response.code = errorCode
            response.message = errorMessage
        }
        return response
    }

    override fun getRoleById(role: Role?): CommonResponce<Role?>? {
        val response = CommonResponce<Role?>()
        try {
            val roleObj: Role? = roleRepo.findById(role?.roleId as Long)
            response.newObject=roleObj
            response.code = successCode
            response.message = successMessage
        } catch (e: java.lang.Exception) {
            response.code = errorCode
            response.message = errorMessage
        }
        return response
    }

    override fun getrolebyRoleCode(role: String?): CommonResponce<Role?>? {
        val response = CommonResponce<Role?>()
        try {
            val roleObj: Role? = roleRepo.getRolebyRoleCode(role as String)
            response.newObject = roleObj
            response.code = successCode
            response.message = successMessage
        } catch (e: java.lang.Exception) {
            response.code = errorCode
            response.message = errorMessage
        }
        return response
    }

    override fun deleteRole(role: Role?): CommonResponce<Role?>? {
        val response = CommonResponce<Role?>()
        try {
            roleRepo.delete(role?.roleId as Long)
            response.code = successCode
            response.message = successMessage
        } catch (e: java.lang.Exception) {
            response.code = errorCode
            response.message = errorMessage
        }
        return response
    }


}